context/user_plane_information.go

```diff
@@ -354,13 +355,77 @@ func (upi *UserPlaneInformation) InsertSmfUserPlaneNode(name string, node *facto
 	return nil
 }
 
-// update UPF info
-func (upi *UserPlaneInformation) UpdateSmfUserPlaneNode(name string, node *factory.UPNode) error {
-	//What all to update- nssai/dnn info
-	logger.UPNodeLog.Infof("UPNode[%v] to update, content[%v]\n", name, node)
-	logger.UPNodeLog.Debugf("content of map[UPNodes] %v \n", upi.UPNodes)
-	upi.DeleteSmfUserPlaneNode(name, node)
-	upi.InsertSmfUserPlaneNode(name, node)
+// Update an existing User Plane Node.
+// If the node is of type AN, then the node is updated with the new port.
+// If the node is of type UPF, then the node is updated with the new port and the new UPF information.
+func (upi *UserPlaneInformation) UpdateSmfUserPlaneNode(name string, newNode *factory.UPNode) error {
+	logger.UPNodeLog.Infof("UPNode [%v] to update, content[%v]\n", name, newNode)
+
+	existingNode, exists := upi.UPNodes[name]
+	if !exists {
+		return fmt.Errorf("UPNode [%v] does not exist", name)
```

Comment:
Given that `names` is string, why not the following?
```suggestion
		return fmt.Errorf("UPNode [%s] does not exist", name)
```